packages/cadl-ranch-specs/http/payload/multipart/mockapi.ts

```diff
@@ -1,19 +1,136 @@
-import { passOnSuccess, ScenarioMockApi, mockapi, ValidationError } from "@azure-tools/cadl-ranch-api";
-import { jpgFile } from "../../helper.js";
+import {
+  passOnSuccess,
+  ScenarioMockApi,
+  mockapi,
+  ValidationError,
+  MockRequest,
+  MockApi,
+  withKeys,
+} from "@azure-tools/cadl-ranch-api";
+import { jpgFile, pngFile } from "../../helper.js";
 
 export const Scenarios: Record<string, ScenarioMockApi> = {};
 
-Scenarios.Payload_MultiPart_FormData_basic = passOnSuccess(
-  mockapi.post("/multipart/form-data/mixed-parts", (req) => {
-    req.expect.deepEqual(req.body.id, "123");
-    if (req.files instanceof Array && req.files?.length > 0) {
-      req.expect.deepEqual(req.files[0].fieldname, "profileImage");
-      req.expect.deepEqual(req.files[0].mimetype, "application/octet-stream");
-      req.expect.deepEqual(req.files[0].buffer, jpgFile);
-    } else {
-      throw new ValidationError("No profileImage found", "jpg file is expected", req.body);
+function checkId(req: MockRequest) {
+  req.expect.deepEqual(req.body.id, "123");
+}
+
+function checkAddress(req: MockRequest) {
+  req.expect.deepEqual(JSON.parse(req.body.address), { city: "X" });
+}
+
+function checkPreviousAddresses(req: MockRequest) {
+  req.expect.deepEqual(JSON.parse(req.body.previousAddresses), [{ city: "Y" }, { city: "Z" }]);
+}
```

Comment:
Why do we need the JSON.parse here? Does that mean the server side may not know this is an application json?

Reply:
`multer` don't help to parse it so we have to do it by ourselves.